### PR TITLE
Fix for "NSExpression function is forbidden"

### PR DESCRIPTION
**Pull request by roblabs**
* For iOS 15.5, refactor how MGL Expressions are defined. Fix for "NSExpression function is forbidden" #331
* Update Swift & Objective-C sample Code for MGL Expressions.  Use these patterns for fixing unit tests.
 Convert CI Scheme to TestPlan. Add CI-Expressions TestPlan (which skips tests unrelated to `NSExpression`)
* Update Swift sample code & documentation for MGL expressions.  Update API docs (.h) & online docs (HTML) with:  `make darwin-style-code && make idocument`
* Refresh developer documentation for:  build dependencies, running unit tests from the command line, and logging in C++.
* Mark certain unit tests as "Expected Failures":  Awaiting unit test refactoring for https://github.com/maplibre/maplibre-gl-native/issues/331
- [ ] TODO:  The unit tests that are "Expected Failures" will need to be refactored.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,7 +169,12 @@ Please refer to [Mapbox Maps SDK for iOS](platform/ios/platform/ios/) for detail
 
 ```bash
 cd platform/ios
+# open macOS project in Xcode
+make xproj
```
Comment (ntadej) [inline README.md:173]: While wrapping commands in a makefile is nice for tests it makes the whole build system even more convoluted in my opinion. We should revisit this in the future.

**Comment by ntadej** [comment]
Hi @roblabs, I looked through the changes and I see no major issues. I commented on the `Makefile` thing which is out of scope of this PR.

I do imagine that you may want to update those tests you disabled before merging to avoid reduced test coverage?

File: README.md (hunk 1 above)
Comment (roblabs) [reply README.md:173]: @ntadej  — did you mean for this comment in `platform/ios/Makefile`.  L172,L173 is for macOS developers (which I had to fix some stuff in.

> While wrapping commands in a makefile is nice for tests it makes the whole build system even more convoluted in my opinion. We should revisit this in the future.

**Comment by roblabs** [comment]
Tadej — yes, there are unit tests & UI tests that need to updated in order for the [`.mm` source code change](https://github.com/maplibre/maplibre-gl-native/pull/411/files#diff-eeb5ffbea5c6e57d3e044b8050e8857756e1f5033f0ab1b73e60b14befc04d0c) to be fully tested for iOS 15.5 (like it was up to iOS 14).  

I have no ETA for these to be complete.  What do think about merging?  I mostly worried about a future merge with the [internal Xcode XML](https://github.com/maplibre/maplibre-gl-native/pull/411/files#diff-efc089d563c3f115223ce8ce02d31c0bfc9f8935946c6f7cee4d7e14d6963425) if this goes too long. 

- [ ] The failing test is something called “_macOS Nitpick source_”.  I checked out what this does, and that particular part of the GitHub workflow/tests should be deprecated.
- [ ] The unit tests that are "Expected Failures" will need to be refactored.

> @ntadej "_I do imagine that you may want to update those tests you disabled before merging to avoid reduced test coverage?_"

---

_iOS "Expected Failures" from the CI scheme_

**Comment by birkskyum** [comment]
Can we somehow fix the macOS CI?

**Review by birkskyum — APPROVED** [review]
I think this PR is great and ready to merge. The unit tests should of course be refactored asap, but I understand the concern about the xcode files that become more problematic the longer it takes to merge. The nitpick seems to only be flagging intentional modifications here, so it makes sense to overrule it as an exception.

**Comment by birkskyum** [comment]
Will the new way to write formats work on older ios versions (<15) as well?

**Review by birkskyum — APPROVED** [review]
I've made an issue #421 to make sure we remember to refactor the unit tests asap so they can be enabled.

**Comment by ntadej** [comment]
The CI seems to still be failing after merge. I know Birk is working on fixing it but we should be a bit more careful in the future :)

**Comment by roblabs** [comment]
Yes, the the expression changes have been tested on iOS 14 & iOS 15.  Please the section in [DEVELOPING.md](https://github.com/maplibre/maplibre-gl-native/blob/main/platform/ios/platform/ios/DEVELOPING.md#testing-from-the-command-line) titled "*Testing from the command line*" which show how to test on other versions of the SDK.

As well, I think there is a way in the `make` subsystem that it is possible to target an iOS SDK.

> @birkskyum  "*Will the new way to write expressions work on older ios versions (<15) as well?*"

**Pull request merged**